lib/takeaway.rb

```diff
@@ -0,0 +1,46 @@
+class Takeaway
+  attr_reader :menu
+
+  def initialize(menu, messager = Messager)
+    @menu = menu
+    @messager = messager
+  end
+
+  def calculate_price(order)
+    order.inject(0) do |sum, item|
+      name = item[0]
+      quantity = item[1]
+      sum + menu.price_of(name) * quantity
+    end
+  end
+
+  def receive_order(order, payment)
+    fail 'Cannot process an empty order' if order.empty?
+    validate_order_items(order)
+    validate_payment(order, payment)
+    messager.send_message(order.customer.phone, create_message)
+  end
+
+  private
+
+  attr_reader :messager
+
+  def validate_order_items(order)
+    order.each do |name, quantity|
+      fail "#{name} is not on the menu" unless menu.include? name
+    end
+  end
+
+  def validate_payment(order, payment)
+    price = calculate_price(order)
+    fail "Incorrect payment: the order total is £%.2f" % price unless payment == price
```

Comment:
Metrics/LineLength: Line is too long. [86/80] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)<br>Style/FormatString: Favor `format` over `String#%`. (https://github.com/bbatsov/ruby-style-guide#sprintf)